### DISCUSSION TITLE
Really fr####n pissed off.

### DISCUSSION THREAD

**Discussion by ghost** [community/community#18687] [Copilot Conversations]
# I do know there is a similar discussion, but I am really REALLY **REALLY** pissed off.
I need Copilot. It's such a pain in the butt without it. Here is a snippet from the Copilot description (from Nightly):
> Once GitHub Copilot is generally available, it will require a subscription. It will be free for students and verified open source maintainers.

At least give it for free to people who had it via the preview. But thank god I joined early enough that it got accepted within a **DAY**. This will seriously hurt many freelancers and some businesses (and me). I am a very young developer (but has a lot of experience - since late 2019).

Here is a quote I made:

> # I am not a student, but the verified open source maintainer probably means I need to be viral.

But I am a open source maintainer. On another platform. This is the F#####G reason I switched away from GH. They don't give a c##p to actual developers, but instead they are selfish. This also why I don't use (the public instance of) GitLab either, and why I use a very innocent [code hosting provider](https://codeberg.org). But LOL, I don't use them anymore (not that they're bad, but it's just more effort) and instead I just host stuff on my OneDrive folder, but I am planning to switch back to CB.

**Comment by luigimannoni** [comment]
I do partly agree with you, does it hurt a developer on a low income and less developed country? Yes. 
Are you young enough? You can snitch off a student license. Are on a low income? Then yes, probably github should keep in mind that option as well.

But does it hurt a developer that is being paid more than the average person does? I don't think so. 
If it does help and speed up your job, then probably you can buy the tool, like all other professional individuals do. 

Have you seen a dentist drilling a teeth using a hand-operated drill? Why designers buy photoshop if they can use a open-source designing tool? As other professionals have other tools coders have theirs as well.

We have plenty of free software already that enables us to sell our products, you can live and work without github copilot, it's an extra rather than an accessibility tool. 

If software engineering themselves are asking for free software from other software engineers then why you should be paid using someone's free work? If you want the free copilot then commit to be an open source maintainer and do free work, it's easy and probably you won't really need to be that much viral just jump on a popular library from someone's else, open the issues tab and start tackling bug and open pull requests.

**Comment by tlhfckoctbcr** [comment]
GitHub is running a business and will likely not part with tools for free, outside of a marketable preview for a short period of time. You should know this from your wealth of experience.

**Reply by NullDev** [reply]
Problem is just they trained their nice little tool on free and open source (and partially closed source) code and now make money from it. 
And obviously the people who involuntarily contributed to the development of said tool don't get anything from the profit GitHub generates with it

**Reply by tlhfckoctbcr** [reply]
It isn't so much a problem when a tool that didn't exist prior can be made available to a market. Thanks to all who contributed, of course.

**Reply by rauzerfabiano** [reply]
@NullDev Exactly!

**Reply by Urethramancer** [reply]
> Problem is just they trained their nice little tool on free and open source

_I_ was trained on free and open source code.

**Reply by Ayushmanchess** [reply]
They should make it free, though. At-least we previewers have it till august 22nd

**Reply by luigimannoni** [reply]
Is there's a paid software which isn't based upon open source tools?

Because by the time you code your software in a language freely available, perhaps also achieve the results by copypasting some of the solutions found freely on the internet and sell said software, then that it is exactly what you're doing. Being paid for putting other people's open source efforts together in such way you can resell it.

**Reply by NullDev** [reply]
@luigimannoni "basing" your tool on FOSS software and adhering to licenses, and "training" your tool on FOSS software without adhering to licenses are two different things

**Reply by luigimannoni** [reply]
@NullDev right, but if you adopt same coding patterns and programming principles found in other code repositories are you infringing their license? Licenses, at least the GPL and a bunch others I know are there to cover only the reproduction, modifications and commercialisation of the licensed software. Nothing is mentioned about learning or training.

Once someone told me "I know you guys at the end just go on stack overflow and copy and paste answers, why we should pay you", I replied "Because I know exactly the right thing I need to copy, paste and re-adapt to make it work".

So if I am doing it already manually then why is it wrong for a machine tool? Shall we also stop using spell/grammar checkers on documents and emails because they have been trained upon unlicensed texts and books?

Image recognition softwares and AIs are trained upon massive amount of, probably, licensed photos. Yet companies charge money to use their software/services due to people's work behind and expensive infrastructure. Same applies to Copilot.

**Reply by tlhfckoctbcr** [reply]
The tool reduces the amount of time spent hunting for solutions and guidance, and anything that saves me as much time as Copilot is worth the cash. The tool didn't exist before, and it does now because of a team of talented engineers at GitHub, and I feel the company offers a fair price for their labor. I agree with @luigimannoni 's perspective. I would also be surprised if GH _didn't_ adhere to FOSS licensing agreements while developing/marketing this new product.

**Comment by orassayag** [comment]
Every paid tool have an open source one. Be sure that free alternatives for copilot will be created.
